src/test/java/com/google/googleidentity/security/LoginCheckServletTest.java

```diff
@@ -0,0 +1,129 @@
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package com.google.googleidentity.security;
+
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.common.base.Charsets;
+import com.google.common.hash.Hashing;
+import com.google.googleidentity.user.InMemoryUserDetailsService;
+import com.google.googleidentity.user.UserDetails;
+import com.google.googleidentity.user.UserDetailsService;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+
+import com.google.inject.Provider;
+
+import org.junit.Test;
+
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test {@link LoginCheckServlet}'s logic
+ */
+public class LoginCheckServletTest  {
+
+    public class UserSessionProvider implements Provider<UserSession> {
+        @Override
+        public UserSession get() {
+            UserSession userSession = new UserSession();
+            userSession.setUser(UserDetails.newBuilder()
+                    .setUsername("user")
+                    .setPassword(Hashing.sha256()
+                            .hashString("123456", Charsets.UTF_8).toString())
+                    .build());
+            return userSession;
+        }
+    }
+
+    @Test
+    public void testLoginCheckServlet_correctLoginRequest_redirectToResource()
+            throws ServletException, IOException {
+
+        UserDetailsService userDetailsService = new InMemoryUserDetailsService();
+
+        userDetailsService.addUser(UserDetails.newBuilder()
+                .setUsername("user")
+                .setPassword(Hashing.sha256()
+                        .hashString("123456", Charsets.UTF_8).toString())
+                .build());
+
+        LoginCheckServlet loginCheckServlet = new LoginCheckServlet(
+                new UserSessionProvider(), userDetailsService);
+
+        HttpServletRequest request = mock(HttpServletRequest.class);
+        HttpServletResponse response = mock(HttpServletResponse.class);
+
+        when(request.getParameter("username")).thenReturn("user");
+        when(request.getParameter("password")).thenReturn(Hashing.sha256()
+                .hashString("123456", Charsets.UTF_8).toString());
+
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter writer = new PrintWriter(stringWriter);
+        when(response.getWriter()).thenReturn(writer);
+
+        loginCheckServlet.doPost(request, response);
+
+        System.out.println(stringWriter.toString());
+
+        assertTrue(stringWriter.toString().equals("/resource/user\r\n"));
```

Comment:
Again, use more specific assertion methods (like assertEquals here) to get better error messages on failures.